nested-data-and-nested-loops/practice-nested-data-and-nested-loops.md

```diff
@@ -0,0 +1,516 @@
+##### !question
+
+Implement `get_student_names()`. This method takes in one parameter, `student_groups`. The method should return a list of strings, which are student names.
+
+Here are the tests:
+
+```python
+def test_get_student_names():
+    students = [
+        ['Michelle', 'Beyonce', 'Kelly'],
+        ['Luke', 'Leia', 'Han'],
+        ['Gordon', 'Lizzie', 'Miranda']
+    ]
+
+    names = get_student_names(students)
+
+    assert names == ['Michelle', 'Beyonce', 'Kelly', 'Luke',
+                     'Leia', 'Han', 'Gordon', 'Lizzie', 'Miranda']
+
+
+def test_get_student_names_no_names():
+    no_names = get_student_names([])
+    assert no_names == []
+```
+
+##### !end-question
+##### !tests
+```python
+import unittest
+from main import *
+
+class TestPython1(unittest.TestCase):
+    def test_get_student_names(self):
+        students = [
+            ['Michelle', 'Beyonce', 'Kelly'],
+            ['Luke', 'Leia', 'Han'],
+            ['Gordon', 'Lizzie', 'Miranda']
+        ]
+        names = get_student_names(students)
+        self.assertEqual(
+            names, ['Michelle', 'Beyonce', 'Kelly', 'Luke', 'Leia', 'Han', 'Gordon', 'Lizzie', 'Miranda'])
+
+    def test_get_student_names_no_names(self):
+        no_names = get_student_names([])
+
+        self.assertEqual(no_names, [])
+```
+##### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def get_student_names(student_groups):
+    names = []
+    for group in student_groups:
+        for student in group:
+            names.append(student)
+    return names
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 5 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: short-answer
+* id: 613ccf11-e7e6-4fad-a1b1-6a962bcd8967
+* title: Nested Data and Nested Loops
+##### !question
+
+What is the syntax to get `'University of Washington'` from `schools`?
+
+```python
+schools = {
+    "school 1": {
+        "name": "Western Washington University",
+        "city": "Bellingham"
+    },
+    "school 2": {
+        "name": "University of Washington",
+        "city": "Seattle"
+    },
+    "school 3": {
+        "name": "Washington State University",
+        "city": "Pullman"
+    }
+}
+```
+
+Use `[]` syntax with double-quotes where needed.
+
+##### !end-question
+##### !answer
+
+schools["school 2"]["name"]
+
+##### !end-answer
+##### !hint
+
+Use `[]` syntax with double-quotes where needed.
+
+Examples:
+
+- `some_list[0]`
+- `some_dictionary["some key"]`
+
+##### !end-hint
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 6 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: b54addb3-48b0-462a-9f6a-f9b749fba39f
+* title: Nested Data and Nested Loops
+##### !question
+
+Implement `get_school_names()`. This method takes in one parameter, `schools`. The method should return a list of strings, which are school names.
```

Comment:
Successful completion of this depends on an implementation detail of CPython 3.6 (formalized in Python 3.7--documented in a note at https://docs.python.org/3/library/stdtypes.html#typesmapping). I notice that the python editors in Learn are at Python 3.6.

Can we assume that's CPython (so that tests like this will pass)?

Can we increase the hosted version if there are language features we encounter that came _after_ 3.6? (Students have 3.9 installed on their machines if they went through installfest.)

Reply:
Similar options as above
```suggestion
Implement `get_school_names()`. This method takes in one parameter, `schools`. The method should return a list of school names, which are strings.
```

or
```suggestion
Implement `get_school_names()`. This method takes in one parameter, `schools`. The method should return a list of strings, which are the school names.
```